### PR TITLE
Return model summary instead of just printing to screen

### PR DESCRIPTION
**Pull request by navdeep-G**
* This PR returns a model summary object (TwoDimTable) in the Python API.
* This will allow a user to manipulate model summaries and get metrics of interest much like what we allow in our R API.

---
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/522)

### CODE REVIEW

File: h2o-py/h2o/model/model_base.py [hunk 1]
```diff
@@ -319,7 +319,7 @@ def summary(self):
         """Print a detailed summary of the model."""
         model = self._model_json["output"]
         if model["model_summary"]:
-            model["model_summary"].show()  # H2OTwoDimTable object
+            return model["model_summary"]
```
Comment (Mistobaan) [inline h2o-py/h2o/model/model_base.py]: what should return if there is no summary ?

**Comment by Mistobaan** [comment]
add a test to confirm that it works

File: h2o-py/h2o/model/model_base.py (hunk 1 above)
Comment (navdeep-G) [inline h2o-py/h2o/model/model_base.py]: Theoretically there is no summary if there is no model. But we can return an exception here that is a bit more informative than a comment that I posted on a PR.

**Comment by navdeep-G** [comment]
@h2o-ops please test!

**Comment by Mistobaan** [comment]
why did it fail ? can't understand the jenkins output

**Comment by navdeep-G** [comment]
Looks like architecture stuff. The main test passed, we can re-run to monitor the situation.

**Comment by navdeep-G** [comment]
@h2o-ops please test!

**Comment by navdeep-G** [comment]
@h2o-ops please test!

**Comment by navdeep-G** [comment]
A lot of non related failures occurring here...

**Comment by navdeep-G** [comment]
@h2o-ops please test

**Comment by mmalohlava** [comment]
@h2o-ops please test!

**Comment by abal5** [comment]
@h2o-ops please test!

**Pull request merged**